subsys/bluetooth/audio/mpl.c

```diff
@@ -1052,20 +1052,13 @@ static void do_first_track(struct mpl_mediaplayer *pl)
 	LOG_DBG_OBJ_ID("Track ID before: ", pl->group->track->id);
 #endif /* CONFIG_BT_MPL_OBJECTS */
 
-	if (pl->group->track->prev != NULL) {
-		pl->group->track = pl->group->track->prev;
-		media_player.track_pos = 0;
-		do_track_change_notifications(&media_player);
-	} else {
-		/* For first track, the position is reset to 0 even */
-		/* if we stay at the same track (goto start of track) */
-		set_track_position(0);
-	}
-
 	while (pl->group->track->prev != NULL) {
 		pl->group->track = pl->group->track->prev;
 	}
 
+	media_player.track_pos = 0;
+	do_track_change_notifications(&media_player);
```

Comment:
This change seems to discard 

```c
/* For first track, the position is reset to 0 even */
/* if we stay at the same track (goto start of track) */
set_track_position(0);
```

So it seems that this new proposed change would always send a track change notification, even if the track doesn't change. That doesn't seem right.

As per your commit message, the issue you are trying to fix seems to be with changing groups, and it's unclear to me why `do_first_track` is the right place to do that.

Reply:
Right... discarded that idea and updated functions related to group change